osu.Game/Database/StandardisedScoreMigrationTools.cs

```diff
@@ -185,6 +187,36 @@ public static long GetOldStandardised(ScoreInfo score)
             return (long)Math.Round((1000000 * (accuracyPortion * accuracyScore + (1 - accuracyPortion) * comboScore) + bonusScore) * modMultiplier);
         }
 
+        public static long ChangeComboRatio(ScoreInfo score, double oldComboPortion, double newComboPortion)
+        {
+            // Assume the incoming score already has a total score in the newest standardised calculation method.
+            double totalScore = score.TotalScore;
+
+            // Undo mod multiplier to leave us with combo and accuracy portions.
+            double scoreMultiplier = 1;
+            foreach (var m in score.Mods)
+                scoreMultiplier *= m.ScoreMultiplier;
+
+            totalScore /= scoreMultiplier;
+
+            var processor = runBestCaseScoreProcessor(score);
+
+            // Remove the accuracy and bonus portions (which we can recalculate)
+            totalScore -= processor.AccuracyPortion * Math.Pow(processor.Accuracy.Value, 10) * (1000000 * (1 - oldComboPortion));
```

Comment:
I take issue with this being placed in `StandardisedScoreMigrationTools` because this particular line only applies for osu! the ruleset. The remaining three do entirely different things with the accuracy portion and as such this method cannot be used for other rulesets.

Reply:
I've renamed the method and enforced the ruleset requirement.

Reply:
Slightly conflicted on that resolution because I foresee similar experiments being done for other rulesets too, in which case 99% of this logic can be reused save for this one specific line with the specific osu!-ruleset implementation detail. But I guess that bridge can be crossed when we come to it.